### PR TITLE
Add and Remove feature added to feature

### PR DESCRIPTION
**Pull request by NatiDeme**
# Changes in this push: 
- No linter errors
- Added add feature
- Added remove feature
- Added edit feature
- All requirements added as listed on project requirement.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -9,7 +9,7 @@
 <body>
     <main>
         <section id="main">
-            <h1>ToDoList</h1>
+            <h1>To Do List</h1>
         </section>
     </main>
 </body>
```
Comment (vic778) [inline src/index.html:16]: - [ ] Kindly you didn't implement any Project requirements;  you should  In this project have the  CRUD (create, update, delete) methods. All the elements of the user interface will be fully functional and your application will be completed.:
   - Remove all hardcoded items from the tasks array.
   - Create a new JavaScript file for the new functionality.
   - Implement a function for adding a new task (add a new element to the array).
   - Implement a function for deleting a task (remove an element from the array).
   - Implement a function for editing task descriptions.
   - By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
   - Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
   - All changes to the To Do List should be saved in local storage.

**Review by vic778 — CHANGES_REQUESTED** [review]
## Hello @NatiDeme 

Good job so far! I know you have worked hard on making this project.  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [x] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/footer.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const footer = () => {
   const footer = document.createElement('div');
   footer.innerHTML += `
-    <p>clear all completed</p>
+    <button class="clearButton" id="clear"><p>clear all completed</p></button>
```
Comment (ahmetbozaci) [inline src/modules/footer.js:4]: ### OPTIONAL
_This feature should add to the last milestone._
- [ ] The clear completed button is not working please fix it. [Requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/sneak_peek.md#:~:text=Removing%20all%20items%20marked%20as%20complete%20at%20once.)

https://user-images.githubusercontent.com/38635158/171861539-dc2fc555-ea81-4fca-aa7e-8c6ee3624aa4.mp4

emoving%20all%20items%20marked%20as%20complete%20at%20once.)

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


@NatiDeme  ✋🏼 
#### Great work on making the changes requested by a previous reviewer 👏 
#### You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
___
### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: src/modules/footer.js (hunk 2 above)
Comment (NatiDeme) [reply src/modules/footer.js:4]: @ahmetbozaci thank you for your review but the clear button is not a requirement for this stage of the project I am submitting for [To Do list: add & remove](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md)

**Comment by nicupop729** [comment]
PR under second opinion review

Kindly, please have patience :smiley_cat:

File: src/modules/footer.js (hunk 2 above)
Comment (ahmetbozaci) [reply src/modules/footer.js:4]: > @ahmetbozaci thank you for your review but the clear button is not a requirement for this stage of the project I am submitting for [To Do list: add & remove](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md)

- [ ] You are right it is not a requirement in this milestone. But it is a [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#:~:text=Implement%20a%20function%20for%20the%20%22Clear%20all%20completed%22%20button%20(use%20filter()%20method).) in the last milestone. You should already did it.
- [ ] For this milestone. I will make it optional.

Comment (ahmetbozaci) [reply src/modules/footer.js:4]: _Next code reviewer, please approve this project._

**Review by nicupop729 — COMMENTED** [review]
Hello @NatiDeme 👋,

_This is an additional review as a result of the second opinion request._

As the reviewer commented earlier, the `Clear all` button was a task that was mandatory for the previous milestone. 
Because this functionality of the app is not from this milestone, it is optional to be implemented now.
In order to have a fully functional app, I strongly suggest you to implement it. Don't forget, that you would like to showcase your skills and your projects to potential employers.
 
As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approve right away :)

Keep rocking 🚀

**Comment by NatiDeme** [comment]
@nicupop729  Thank you for the second opinion. It was really helpful.

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @NatiDeme 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 
✅ Descriptive PR 👍 
✅ Nice design 🔥 
✅ Linters are passing 👏 


## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**